provisioning/service/src/Auth/IAuthenticationMethod.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Provisioning.Service
+{
+    /// <summary>
+    /// Authentication interface to use for IoTHub communications.
+    /// </summary>
+    public interface IAuthenticationMethod
```

Comment:
We already have 1 or even 2 copies of the same public type. We should think of moving it to Shared.

Reply:
The entire Auth should be move to Common or shared.

Reply:
We will postponed it to a next PR